pixeltable/catalog/table_version.py

```diff
@@ -697,7 +684,24 @@ def _add_columns(
         self, cols: Iterable[Column], print_stats: bool, on_error: Literal['abort', 'ignore']
     ) -> UpdateStatus:
         """Add and populate columns within the current transaction"""
+        from pixeltable.catalog import Catalog
+        from pixeltable.plan import Planner
+
         cols_to_add = list(cols)
+
+        @env.register_rollback_action
```

Comment:
This isn't necessary, you already mark this instance is modified.

But would be good to do this here explicitly, since mark_modified_tvs() is idempotent, but it's critical to have that here.

Reply:
Ok, so it isn't necessary to call `self.store_tbl.create_sa_tbl()` at the end? I thought that was the point of the undo action.